platform/pulpcore/app/models/content.py

```diff
@@ -38,36 +76,47 @@ def natural_key(self):
+
+
+class DeferredArtifact(Model):
     """
-    def storage_path(self, name):
-        """
-        Callable used by FileField to determine where the uploaded file should be stored.
+    Stores information about an artifact that has not been retrieved yet.
 
-        Args:
-            name (str): Original name of uploaded file. It is ignored by this method because the
-                sha256 checksum is used to determine a file path instead.
-        """
-        return default_storage.get_artifact_path(self.sha256)
+    Importers that want to support deferred download policies should use this model to store
+    information required for downloading an Artifact at some point in the future. At a minimum this
+    includes the URL, the ContentArtifact, and the Importer that created it. It can also store
+    expected size and any expected checksums.
 
-    file = models.FileField(db_index=True, upload_to=storage_path, max_length=255)
-    size = models.IntegerField(blank=False, null=False)
-    md5 = models.CharField(max_length=32, blank=False, null=False, unique=False, db_index=True)
-    sha1 = models.CharField(max_length=40, blank=False, null=False, unique=False, db_index=True)
-    sha224 = models.CharField(max_length=56, blank=False, null=False, unique=False, db_index=True)
-    sha256 = models.CharField(max_length=64, blank=False, null=False, unique=True, db_index=True)
-    sha384 = models.CharField(max_length=96, blank=False, null=False, unique=True, db_index=True)
-    sha512 = models.CharField(max_length=128, blank=False, null=False, unique=True, db_index=True)
+    Fields:
+
+        url (models.TextField): The URL where the artifact can be retrieved.
+        size (models.IntegerField): The expected size of the file in bytes.
+        md5 (models.CharField): The expected MD5 checksum of the file.
+        sha1 (models.CharField): The expected SHA-1 checksum of the file.
+        sha224 (models.CharField): The expected SHA-224 checksum of the file.
+        sha256 (models.CharField): The expected SHA-256 checksum of the file.
+        sha384 (models.CharField): The expected SHA-384 checksum of the file.
+        sha512 (models.CharField): The expected SHA-512 checksum of the file.
+    """
+    url = models.TextField(blank=True, validators=[validators.URLValidator])
+    size = models.IntegerField(blank=True, null=True)
+    md5 = models.CharField(max_length=32, blank=True, null=True)
+    sha1 = models.CharField(max_length=40, blank=True, null=True)
+    sha224 = models.CharField(max_length=56, blank=True, null=True)
+    sha256 = models.CharField(max_length=64, blank=True, null=True)
+    sha384 = models.CharField(max_length=96, blank=True, null=True)
+    sha512 = models.CharField(max_length=128, blank=True, null=True)
+    content_artifact = models.ForeignKey(ContentArtifact, on_delete=models.CASCADE)
+    importer = models.ForeignKey('Importer', on_delete=models.CASCADE)
```

Comment:
Relations should be added to the docstring (as per [style guide](http://docs.pulpproject.org/en/3.0/nightly/contributing/style_guide.html#exceptions-and-clarifications)).